frontend/src/components/matching-service/StartMatchingForm.tsx

```diff
@@ -37,14 +38,14 @@ export default function StartMatchingForm() {
     const topicsStr = questionTopics.join(", ");
     navigate(`../matching/wait?difficulties=${difficultiesStr}&topics=${topicsStr}`);
 
-    sendStartMatchingRequest(auth.token, selectedDifficultyData, questionTopics).then(
+    sendStartMatchingRequest(auth.id, auth.email, selectedDifficultyData, questionTopics).then(
       response => {
         const httpStatus = response.status;
         const errorMessage = response.message
 
         if (httpStatus === HTTP_OK) {
           // navigate("/matching/get_ready")
-        } else if (httpStatus === HTTP_NO_CONTENT || httpStatus === HTTP_REQUEST_TIMEOUT) {
+        } else if (httpStatus === HTTP_NO_CONTENT || httpStatus === HTTP_REQUEST_TIMEOUT || httpStatus === HTTP_ALREADY_EXISTS) {
           navigate(`/matching/failed?message=${errorMessage}&difficulties=${difficultiesStr}&topics=${topicsStr}`);
         } else {
           displayError("An error has occured: \n" + response.message);
```

Comment:
this part seems a lil confusing, perhaps can cleanup (in another PR?). The http statuses need to be coordinated with those in matching-service/routes. Particularly in 

`else if (httpStatus === HTTP_NO_CONTENT || httpStatus === HTTP_REQUEST_TIMEOUT || httpStatus === HTTP_ALREADY_EXISTS)`

idt theres a need to check for so many statuses

Reply:
i think it's fine as it is

Reply:
i think it's fine as it is

Reply:
i say its confusing cos

1. `if (httpStatus === HTTP_OK) {
          // navigate("/matching/get_ready")`

this part is redundant

2. in her current implementation in `matching-service/routes`, http status 204 and 408 is never returned, so why wd u keep `HTTP_NO_CONTENT` and `HTTP_REQUEST_TIMEOUT`?